src/app/main/component/events/components/event-editor/components/create-event-dates/components/date-time/date-time.component.ts

```diff
@@ -116,6 +116,22 @@ export class DateTimeComponent implements OnInit, AfterViewInit {
     this.endTime.valueChanges.subscribe((value: string) => {
       this._handleTimeChange(value, 'end');
     });
+
+    // Subscribe to date value changes
+    this.date.valueChanges.subscribe((newDate) => {
+      this._updateNeighboringDates();
+
+      if (!newDate) {
```

Comment:
To avoid nested if statements, you can use a switch statement to handle different conditions.

```
      let result = null;
      switch (true) {
        case !newDate:
          result = { dateIncorrect: true };
          break;
        case newDate < new Date():
          result = { dateInPast: true };
          break;
      }
       this.date.setErrors(result);
```